neqo-transport/src/tracking.rs

```diff
@@ -93,10 +95,49 @@ impl SentPacket {
         }
     }
 
+    /// Returns `true` if the packet counts requires congestion control accounting.
+    /// The specification uses the term "in flight" for this.
+    pub fn cc_in_flight(&self) -> bool {
+        self.in_flight
+    }
+
+    /// Whether the packet has been declared lost.
+    pub fn lost(&self) -> bool {
+        self.time_declared_lost.is_some()
+    }
+
+    /// Whether accounting for the loss or acknowledgement in the
+    /// congestion controller is pending.
+    /// Returns `true` if the packet counts as being "in flight",
+    /// and has not previously been declared lost.
+    pub fn cc_outstanding(&self) -> bool {
+        self.cc_in_flight() && !self.lost()
+    }
+
+    /// Declare the packet as lost.  Returns `true` if this is the first time.
+    pub fn declare_lost(&mut self, now: Instant) -> bool {
+        if self.lost() {
+            false
+        } else {
```

Comment:
We hit this function multiple times.  As you might recall, we have cases where we mark a packet as lost, but keep it around just in case it is acknowledged.  During that time, we will keep hitting this function.  Asserting here affects about 10 test cases, with a variety of cases, so it looks like we depend on this path occasionally.